front-end/src/app/transactions/transaction-group-b/transaction-group-b.component.ts

```diff
@@ -1,8 +1,158 @@
+
+  formProperties: string[] = [
+    'entity_type',
+    'contributor_organization_name',
+    'contributor_last_name',
+    'contributor_first_name',
+    'contributor_middle_name',
+    'contributor_prefix',
+    'contributor_suffix',
+    'contributor_street_1',
+    'contributor_street_2',
+    'contributor_city',
+    'contributor_state',
+    'contributor_zip',
+    'contribution_date',
+    'contribution_amount',
+    'contribution_aggregate',
+    'contribution_purpose_descrip',
+    'memo_code',
+    'memo_text_description',
+  ];
+  ContactTypes = ContactTypes;
+  contactTypeOptions: PrimeOptions = LabelUtils.getPrimeOptions(ContactTypeLabels);
+  stateOptions: PrimeOptions = LabelUtils.getPrimeOptions(LabelUtils.getStateCodeLabelsWithoutMilitary());
+  destroy$: Subject<boolean> = new Subject<boolean>();
+  formSubmitted = false;
+
+  form: FormGroup = this.fb.group({});
+
+  constructor(
+    private messageService: MessageService,
+    private transactionService: TransactionService,
+    private validateService: ValidateService,
+    private fb: FormBuilder,
+    private router: Router
+  ) {}
+
+  ngOnInit(): void {
+    // Intialize FormGroup, this must be done here. Not working when initialized only above the constructor().
+    this.form = this.fb.group(this.validateService.getFormGroupFields(this.formProperties));
+
+    // Initialize validation tracking of current JSON schema and form data
+    this.validateService.formValidatorSchema = this.schema;
+    this.validateService.formValidatorForm = this.form;
+
+    // Intialize form on "Individual" entity type
+    if (this.transaction?.id) {
+      const txn = { ...this.transaction } as SchATransaction;
+      this.form.patchValue({ ...txn });
+      this.form.patchValue({ contribution_date: DateUtils.convertFecFormatToDate(txn.contribution_date) });
+    } else {
+      this.form.patchValue({
+        entity_type: ContactTypes.INDIVIDUAL,
+        contribution_aggregate: '0',
+      });
+    }
+
+    this.form
+      ?.get('entity_type')
+      ?.valueChanges.pipe(takeUntil(this.destroy$))
+      .subscribe((value: string) => {
+        if ([ContactTypes.INDIVIDUAL, ContactTypes.CANDIDATE].includes(value as ContactTypes)) {
+          this.form.patchValue({
+            contributor_organization_name: '',
+          });
+        }
+        if ([ContactTypes.ORGANIZATION, ContactTypes.COMMITTEE].includes(value as ContactTypes)) {
+          this.form.patchValue({
+            contributor_last_name: '',
+            contributor_first_name: '',
+          });
+        }
+      });
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next(true);
+    this.destroy$.complete();
+  }
+
+  save(navigateTo: 'list' | 'add another') {
+    this.formSubmitted = true;
+
+    if (this.form.invalid) {
+      return;
+    }
+
+    // Force selection of checkbox to true/false even if never checked
+    this.form.patchValue({
+      memo_code: !!this.form.get('memo_code')?.value,
```

Comment:
do we get the same functionality if we set the memo_code to false by default in the template?

Reply:
For the p-checkbox that is set to binary mode ([binary]="true") there is no attribute to set in the template to make the default  value "false", it is returning an empty string. I updated the form initialization to patchValue a "false" value and moved that force patch value out of the save() method.